src/libcore/slice/mod.rs

```diff
@@ -834,13 +834,47 @@ impl<T> [T] {
     #[stable(feature = "chunks_exact", since = "1.31.0")]
     #[inline]
     pub fn chunks_exact_mut(&mut self, chunk_size: usize) -> ChunksExactMut<'_, T> {
-        assert!(chunk_size != 0);
+        assert_ne!(chunk_size, 0);
         let rem = self.len() % chunk_size;
         let len = self.len() - rem;
         let (fst, snd) = self.split_at_mut(len);
         ChunksExactMut { v: fst, rem: snd, chunk_size }
     }
 
+    /// Returns an iterator over `N` elements of the slice at a time, starting at the
+    /// beginning of the slice.
+    ///
+    /// The chunks are slices and do not overlap. If `N` does not divide the length of the
+    /// slice, then the last up to `N-1` elements will be omitted and can be retrieved
+    /// from the `remainder` function of the iterator.
```

Comment:
Could we mention that this is equivalent to `chunks`? Not specifying this might let people get the wrong idea, `chunks_exact`. Maybe we should also have a version for that?

Reply:
This is actually equivalent to `chunks_exact`, as it returns `&[T; N]`, which must always contain exactly N elements.

Because of this, implementing `chunks` usings arrays/const generics doesn't really make sense here.

Reply:
Oh, then how about a version for `chunks`? If there is a generic version for `chunks_exact` I think there should be one for `chunks` too.

If we took `chunks_array` then the `chunks` version will be `chunks_array_not_exact`? Or maybe we should keep the `_exact` for this?